app/src/component/AppNavigator.js

```diff
@@ -128,12 +128,14 @@ class AppWithNavigationState extends React.Component {
       const { dispatch, nav } = this.props;
       // There should be a better way to check this
       if (nav.routes.length > 0) {
-        const subRoutes = nav.routes[0].routes[0].routes;
-        const lastRoute = subRoutes[subRoutes.length - 1];
-        if (nav.routes[0].routes[0].index > 0 &&
-            ['About', 'Settings'].indexOf(lastRoute.key) > -1) {
-          dispatch(NavigationActions.back());
-          return true;
+        if (nav.routes[0].routes && nav.routes[0].routes.length > 0) {
+          const subRoutes = nav.routes[0].routes[0].routes;
+          const lastRoute = subRoutes[subRoutes.length - 1];
+          if (nav.routes[0].routes[0].index > 0 &&
+              ['About', 'Settings'].indexOf(lastRoute.key) > -1) {
```

Comment:
Yes, I agree. However, since we have a fixed navigation structure which is unlikely to change often, I think it's a passable solution until we can come up with a better approach (see https://github.com/lbryio/lbry-android/blob/f8af04ba4b20162d714501acdcf59dcb829f9f41/app/src/component/AppNavigator.js#L129).

Reply:
Setting aside the specific code choices made here, I wanted to comment on the more general situation of how we handle shipping mediocre/bad temporary solutions (which is part of normal development and not something that has to be avoided 100%).

If we're writing code that we know is subpar pending another change or an issue, we should:

1) Make sure that issue is filed ("Proper Routing Architecture" or whatever the more fundamental issue here is)
2) Make sure that issue is linked in code and that the issue links to the code

Coming back to this specific choice, if we must do things this way, it could make sense to hoist the declaration up to the top of the file to add clarity about it's existence.